### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Mr-Craig**
- Reimplemented MeasureStringWidth
- UIList now accepts mouse inputs
- Fixed vertical and horizontal blurry grids.
- Unclosable menu bug Issue #81 
- UIColourPanel style seperator
- onActive reimplemented Issue #97 
- Fix subtitle colouring
- DisplayPageCounter(bool) implemented

### CODE REVIEW

File: menu/items/UIButton.lua [hunk 1]
```diff
@@ -107,6 +107,10 @@ function RageUI.Button(Label, Description, Style, Enabled, Callback, Submenu)
                                 RageUI.NextMenu = Submenu
                             end
                         end
+                    elseif Selected then
+                        if(Callback.onActive ~= nil) then
+                            Citizen.CreateThread(Callback.onActive)
```
Comment (iTexZoz) [inline menu/items/UIButton.lua:112]: Running this in another thread may create a problem with the menu commands.

File: menu/items/UIList.lua [hunk 2]
```diff
@@ -152,6 +152,10 @@ function RageUI.List(Label, Items, Index, Description, Style, Enabled, Actions,
                     if Submenu ~= nil and type(Submenu) == "table" then
                         RageUI.NextMenu = Submenu[Index]
                     end
+                elseif Selected then
+                    if(Actions.onActive ~= nil) then
+                        Citizen.CreateThread(Actions.onActive)
```
Comment (iTexZoz) [inline menu/items/UIList.lua:157]: Running this in another thread may create a problem with the menu commands.

File: menu/items/UISlider.lua [hunk 3]
```diff
@@ -154,6 +154,10 @@ function RageUI.Slider(Label, ProgressStart, ProgressMax, Description, Divider,
                         Actions.onSelected(ProgressStart);
                     end
                     RageUI.PlaySound(Audio[Audio.Use].Select.audioName, Audio[Audio.Use].Select.audioRef)
+                elseif Selected then
+                    if(Actions.onActive ~= nil) then
+                        Citizen.CreateThread(Actions.onActive)
+                    end 
                 end
             end
```
Comment (iTexZoz) [inline menu/items/UISlider.lua:162]: Running this in another thread may create a problem with the menu commands.

File: menu/items/UISliderHeritage.lua [hunk 4]
```diff
@@ -105,6 +105,10 @@ function RageUI.UISliderHeritage(Label, ItemIndex, Description, Actions, Value)
                         Actions.onSelected(ItemIndex / 10, ItemIndex);
                     end
                     RageUI.PlaySound(Audio[Audio.Use].Select.audioName, Audio[Audio.Use].Select.audioRef, false)
+                elseif Selected then
+                    if(Actions.onActive ~= nil) then
+                        Citizen.CreateThread(Actions.onActive)
```
Comment (iTexZoz) [inline menu/items/UISliderHeritage.lua:110]: Running this in another thread may create a problem with the menu commands.

File: menu/items/UISliderProgress.lua [hunk 5]
```diff
@@ -175,6 +175,10 @@ function RageUI.SliderProgress(Label, ProgressStart, ProgressMax, Description, S
                         Actions.onSelected(ProgressStart);
                     end
                     RageUI.PlaySound(Audio[Audio.Use].Select.audioName, Audio[Audio.Use].Select.audioRef)
+                elseif Selected then
+                    if(Actions.onActive ~= nil) then
+                        Citizen.CreateThread(Actions.onActive)
```
Comment (iTexZoz) [inline menu/items/UISliderProgress.lua:180]: Running this in another thread may create a problem with the menu commands.

File: menu/panels/UIGridPanel.lua [hunk 6]
```diff
@@ -20,8 +20,8 @@ local GridType = RageUI.Enum {
 
 local GridSprite = {
     [GridType.Default] = { Dictionary = "pause_menu_pages_char_mom_dad", Texture = "nose_grid", },
-    [GridType.Horizontal] = { Dictionary = "RageUI", Texture = "horizontal_grid", },
-    [GridType.Vertical] = { Dictionary = "RageUI", Texture = "vertical_grid", },
+    [GridType.Horizontal] = { Dictionary = "RageUI_", Texture = "horizontal_grid", },
```
Comment (iTexZoz) [inline menu/panels/UIGridPanel.lua:23]: Can you give me an explanation ?

File: menu/panels/UIGridPanel.lua [hunk 7]
```diff
@@ -20,8 +20,8 @@ local GridType = RageUI.Enum {
...
+    [GridType.Vertical] = { Dictionary = "RageUI_", Texture = "vertical_grid", },
```
Comment (iTexZoz) [inline menu/panels/UIGridPanel.lua:24]: Can you give me an explanation?

Comment (Mr-Craig) [reply menu/panels/UIGridPanel.lua:24]: I changed the mipmap level to 1 through OpenIV and renaming the file was the only way to get the game to restream the data

Comment (iTexZoz) [reply menu/panels/UIGridPanel.lua:24]: Okey, perfect

File: menu/items/UISliderProgress.lua (hunk 5 above)
Comment (Mr-Craig) [reply menu/items/UISliderProgress.lua]: Fixed

**Comment by iTexZoz** [comment]
Perfect, Great stuff 👍

**Pull request merged**